### PR TITLE
[tosa] Add support for torch.aten.upsample_bilinear2d legalization

### PR DESCRIPTION
**Pull request by Tessil**
Add TOSA legalization support for torch.aten.upsample_bilinear2d sharing the resize legalization logic in `convertResizeOp`.

### CODE REVIEW

File: lib/Conversion/TorchToTosa/TosaLegalizeUtils.cpp [hunk 1]
```diff
@@ -585,5 +586,37 @@ bool typeHasZeroDim(ShapedType type) {
...
+  scaleD = applyAligned ? (inputSize - 1) : inputSize;
+
+  // Simplify the scalers, make sure they are even values.
+  int gcd = std::gcd(scaleN, scaleD);
+  scaleN = 2 * scaleN / gcd;
+  scaleD = 2 * scaleD / gcd;
+
+  // If nearest neighbors we need to guarantee we round up.
+  offset = 0;
+  if (mode == tosa::ResizeMode::NEAREST_NEIGHBOR && alignCorners) {
```
Comment (justin-ngo-arm) [inline lib/Conversion/TorchToTosa/TosaLegalizeUtils.cpp:613]: These resize params calculations might be out of date, as they don't match the examples in the TOSA spec: https://www.mlplatform.org/tosa/tosa_spec_1_0_1.html#_resize. Also, according to PyTorch's interpolate spec, align_corners doesn't apply to nearest_neighbor mode, so this check might be outdated as well https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.interpolate.html

cc: @sjarus as the author of the original calculations

Comment (sjarus) [reply lib/Conversion/TorchToTosa/TosaLegalizeUtils.cpp:613]: Since this lines up with what we've updated intentionally, this looks fine to me! As Justin mentions, please make sure the two are in sync.

Comment (Tessil) [reply lib/Conversion/TorchToTosa/TosaLegalizeUtils.cpp:613]: Thanks for the info. 

For now I have kept the behaviour of torch.aten.upsample_bilinear2d legalization the same as `torch.aten.__interpolate` with bilinear resize (the `computeResizeParams` is the same as the `normalize` lambda) and avoided changing anything related to nearest neighbour resize.

Should we update this in this PR or eventually a second separate one? I would generally prefer a second one if it affects the nearest neighbour resize.

Comment (sjarus) [reply lib/Conversion/TorchToTosa/TosaLegalizeUtils.cpp:613]: Whatever is mechanically ideal is fine with me :-)

Comment (justin-ngo-arm) [reply lib/Conversion/TorchToTosa/TosaLegalizeUtils.cpp:613]: A second PR for nearest neighbor resize fix would be my preferred option too
